sys/fmt/table.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_fmt_table
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the table extensions of the string formatting
+ *              library
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <stdarg.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <string.h>
+
+#include "fmt.h"
+
+const char fmt_table_spaces[16] = "                ";
```

Comment:
It feels weird to add 16 bytes of spaces to ROM. How about just replacing ```print_pattern(fmt_table_spaces)``` with ```size_t fill_size = foo; while(foo--) { print(" ", 1) }```, and drop fmt_table_spaces and print_pattern()?

It probably saves ROM.

Reply:
or just use ```memset(out, ' ', fill_size);```?

Reply:
> or just use `memset(out, ' ', fill_size);`?

(ah, no, not writing to a buffer)

Reply:
This allows to have one call for every 16 spaces to `write()` (called by `print()`), instead of 16 calls. If the UART implementation is using e.g. DMA, this could allow the MCU to got to a lower power state for 16 times longer.

Reply:
Hm, I don't agree at all. Then again, as long as this is an implementation detail that someone can pick up later, I don't mind.

Would you be OK with making fmt_table_spaces and print_pattern static and removing it from the public header?

Reply:
Done